src/components/ProjectSettingsPage/index.jsx

```diff
@@ -103,66 +111,61 @@ class ProjectSettingsPage extends React.Component {
   }
 
   handleSubmit() {
-    const { projectName, projectDescription } = this.state;
+    const { projectName, projectDescription,projectOrganisation,projectType, otherType, othersBool } = this.state;
     const {
       updateProject,
       match: {
         params: { projectID },
       },
     } = this.props;
+    const type = othersBool ? (otherType) : (projectType);
     const projectInfo = JSON.parse(localStorage.getItem("project"));
-    const { name, description } = projectInfo;
+    const { name, description, organisation, project_type } = projectInfo;
 
-    const trimmed = (input) => input.trim();
-    const trimprojectName = trimmed(projectName);
-    const trimprojectDescription = trimmed(projectDescription);
+  
 
-    if (trimprojectName !== name || trimprojectDescription !== description) {
-      if (!trimprojectName || !trimprojectDescription) {
+    const Trim = (input) => input.trim();
+    const trimedprojectName = Trim(projectName);
+    const trimedprojectDescription = Trim(projectDescription);
+    const trimedprojectOrganisation = Trim(projectOrganisation);
+    const trimedprojectType = Trim(type)
+
+    if (trimedprojectName !== name || trimedprojectDescription !== description || trimedprojectOrganisation !== organisation || trimedprojectType !== project_type) {
+      if (!trimedprojectName || !trimedprojectDescription || !trimedprojectOrganisation || !trimedprojectType) {
         this.setState({
-          error: "Please provide either a new name or description",
+          error: "Can't update when an empty field is submited, please fill the missing field or leave it unchanged.",
         });
       } else {
-        if (
-          trimprojectName !== name &&
-          trimprojectDescription === description
-        ) {
-          const nameCheckResult = this.checkProjectName(trimprojectName);
+        if ( trimedprojectName !== name ) {
+          const nameCheckResult = this.checkProjectName(trimedprojectName);
           if (nameCheckResult !== "") {
             this.setState({
               error: nameCheckResult,
             });
           } else {
-            const newProject = { name: trimprojectName };
+            //update is successfull whether other fields apart from mame are not changed or not 
+            // as long as they are not empty
+            const newProject = { 
+              name: trimedprojectName,
+              project_type: trimedprojectType,
+              organisation: trimedprojectOrganisation,
```

Comment:
Can you kindly use a regex to filter out most symbols and only allow hyphens, looks like the backend did not have validation we ought to enforce because we will have symbols as organisations like below and that is not valid.

![Screenshot from 2021-11-02 04-13-43](https://user-images.githubusercontent.com/32802973/139771476-f52592a8-8ad1-4e5b-944f-649e7811f724.png)

Reply:
Sure, on it.